app/content.ts

```diff
@@ -0,0 +1,75 @@
+import * as $ from 'jquery'
+interface requestObject{
+  url:string;
+  method:string;
+  requestId:number,
+  timeStamp:number
+}
+class Intercept {
+  requestDetail: object;
+  constructor() {
+    this.requestDetail = {};
+    this.startMessageListener = this.startMessageListener.bind(this);
+    this.injectScripts = this.injectScripts.bind(this);
+    this.initScript = this.initScript.bind(this);
+    this.injectScripts();
+  }
+  startMessageListener = () => {
+    chrome.runtime.onMessage.addListener((request, _, __) => {
+      if (request.message == "INTERCEPT_REQUEST") {
+        this.initScript(request);
+      }
+    });
+  };
+  injectScripts = () => {
+    var jquery = document.createElement("script");
```

Comment:
Avoid mixing var and rely on const and let. var has hoisting properties very different from the other two.